packages/core/src/config/config-resolvers.ts

```diff
@@ -151,6 +151,9 @@ export function resolvePlugins(
         if (pluginModule.rules.oas2) {
           plugin.rules.oas2 = prefixRules(pluginModule.rules.oas2, id);
         }
+        if (pluginModule.rules.async2) {
+          plugin.rules.async2 = prefixRules(pluginModule.rules.async2, id);
+        }
       }
       if (pluginModule.preprocessors) {
         if (!pluginModule.preprocessors.oas3 && !pluginModule.preprocessors.oas2) {
```

Comment:
Shouldn't we allow also async preprocessors? 

```suggestion
        if (!pluginModule.preprocessors.oas3 && !pluginModule.preprocessors.oas2 && !pluginModule.preprocessors.async2) {
```

Reply:
For some reason I cannot select the line 161, but it also needs to be updated, I believe.

Reply:
Similar on lines 144-145 and 177-178.

Reply:
@RomanHotsiy can you comment on this one please?

Reply:
We don't have to.

We can always start from "do not allow" and allow later. But I don't see any problems with allowing them.

Reply:
Got it. I've added the support @tatomyr described, I can't see a reason not to add this either.